tooling/cli/lib/deploy.js

```diff
@@ -196,31 +195,43 @@ async function triggerBatch(instance, key) {
 }
 
 async function getHeliumUploadData(instance) {
-  const endpoint = instanceEndpoint(instance);
-
-  const options = {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json' },
-    body: JSON.stringify({ query: KEY_QUERY, variables: { nickname: instance.nickname } })
-  };
-
-  let responseData = {};
-  const launchData = await fetch(endpoint, options).then(r => r.json());
-
-  if (launchData && launchData.data) {
-    const {
-      data: {
-        HeliumLaunchData: { key, signedUrl }
-      }
-    } = launchData;
+  return new Promise((resolve, reject) => {
+    const endpoint = instanceEndpoint(instance);
 
-    responseData = {
-      key,
-      signedUrl
+    const options = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ query: KEY_QUERY, variables: { nickname: instance.nickname } })
     };
-  }
 
-  return responseData;
+    let responseData = {};
+    fetch(endpoint, options)
+      .then(r => r.json())
+      .then(res => {
+        const resObj = res;
+        if (resObj && resObj.data.HeliumLaunchData) {
+          //there is still a null object in the data
+          const {
+            data: {
+              HeliumLaunchData: { key, signedUrl }
+            }
+          } = resObj;
+
+          responseData = {
+            key,
+            signedUrl
+          };
+          resolve(responseData);
+        } else if (resObj && resObj.errors) {
+          const err = resObj.errors[0];
+          reject(err.message);
+        }
+        resolve(responseData);
```

Comment:
Don't need this `resolve` here since it'll either hit `resolve` or `reject` in the `if/else`.